widgets/edd-widget-most-commented.php

```diff
@@ -1,265 +1,257 @@
-                    <label for="<?php echo $this->get_field_id( 'show_price' ); ?>"><?php _e( 'Display price?', 'edd-widgets-pack' ); ?></label> 
-                </p>                
-                <p>
-                    <input id="<?php echo $this->get_field_id( 'thumbnail' ); ?>" name="<?php echo $this->get_field_name( 'thumbnail' ); ?>" type="checkbox" value="1" <?php checked( '1', $thumbnail ); ?>/>
-                    <label for="<?php echo $this->get_field_id( 'thumbnail' ); ?>"><?php _e( 'Display thumbnails?', 'edd-widgets-pack' ); ?></label> 
-                </p>
-                <p>
-                    <label for="<?php echo $this->get_field_id( 'thumbnail_size' ); ?>"><?php _e( 'Size of the thumbnails, e.g. <em>80</em> = 80x80px', 'edd-widgets-pack' ); ?></label> 
-                    <input class="widefat" id="<?php echo $this->get_field_id( 'thumbnail_size' ); ?>" name="<?php echo $this->get_field_name( 'thumbnail_size' ); ?>" type="text" value="<?php echo $thumbnail_size; ?>" />
-                </p>
-            <?php
-        }
-
-    }
+						// get the post thumbnail.
+						if ( 1 === $thumbnail && has_post_thumbnail( $download->ID ) ) {
+							$post_thumbnail = get_the_post_thumbnail( $download->ID, $thumbnail_size, array( 'title' => esc_attr( $title_attr ) ) ) . "\n";
+							$out .= "<li class=\"widget-download-with-thumbnail\">\n";
+							$out .= sprintf( $link, get_permalink( $download->ID ), esc_attr( $title_attr ), 'widget-download-thumb', $post_thumbnail );
+						} else {
+							$out .= "<li>\n";
+						}
+
+						// append the download's title.
+						$out .= sprintf( $link, get_permalink( $download->ID ), esc_attr( $title_attr ), 'widget-download-title', $title );
+
+						// get the price.
+						if ( 1 === $show_price ) {
+							if ( edd_has_variable_prices( $download->ID ) ) {
+								$price = edd_price_range( $download->ID );
+							} else {
+								$price = edd_currency_filter( edd_get_download_price( $download->ID ) );
+							}
+							$out .= sprintf( "<span class=\"widget-download-price\">%s</span>\n", $price );
+						}
+
+						// finish this element.
+						$out .= "</li>\n";
+					}
+					// finish the list.
+					$out .= "</ul>\n";
+				}
+
+				// set the widget's containers.
+				$cache = $args['before_widget'] . $out . $args['after_widget'];
+
+				// store the result on a temporal transient.
+				set_transient( 'edd_widgets_most_commented', $cache );
+
+			}
+
+			echo $cache;
+
+		}
+
+		/**
+		 * Update
+		 *
+		 * @return   array
+		 * @since    1.0
+		 */
+		public function update( $new_instance, $old_instance ) {
+			$instance = $old_instance;
+
+			// sanitize title.
+			$instance['title'] = strip_tags( $new_instance['title'] );
+
+			// sanitize limit.
+			$instance['limit'] = isset( $instance['limit'] ) ? (int) $instance['limit'] : 4;
```

Comment:
```suggestion
			$instance['limit'] = isset( $new_instance['limit'] ) ? (int) $new_instance['limit'] : 4;
```
Needs to check against the `$new_instance`.